astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +187,59 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(
+    PyUnitListProxy *a,
+    PyUnitListProxy *b,
+    int op) {
+
+  int equal;
+  int status;
+  struct PyUnitListProxy *ax;
+  struct PyUnitListProxy *bx;
+   
+  
+  if ((op == Py_EQ || op == Py_NE) &&
+      PyObject_TypeCheck(b, &PyUnitListProxyType)) {
+      ax = &((PyUnitListProxy *)a)->pyobject;
+      bx = &((PyUnitListProxy *)b)->pyobject;
+          
+    
+    if (!PyObject_RichCompareBool(a->unit_class, b->unit_class,Py_EQ)||
```

Comment:
You have to check the `PyObject_RichCompareBool` for error (-1), otherwise we will could get `SystemError`s.

Reply:
I would prefer a positive check here, e.g. `PyObject_RichCompareBool && size_equal && char_arrays_equal` => `equal = 1`, otherwise `equal = 0`, makes it easier to understand, feel free to ignore that if you think otherwise